tests/imap/test_pooling.py

```diff
@@ -34,11 +33,10 @@ def test_pool():
     assert conn in get_all(pool._queue)
 
 
-@pytest.mark.skipif(True, reason="randomly blocks forever")
-def test_block_on_depleted_pool():
+def test_timeout_on_depleted_pool():
     pool = TestableConnectionPool(1, num_connections=1, readonly=True)
-    # Test that getting a connection when the pool is empty blocks
-    with pytest.raises(gevent.hub.LoopExit), pool.get(), pool.get():
+    # Test that getting a connection when the pool is empty times out
+    with pytest.raises(ConnectionPoolTimeoutError), pool.get(), pool.get(timeout=0.1):
         pass
```

Comment:
Previously this relied on an implementation detail of gevent. `gevent.hub.LoopExit` is internal gevent exception that you are not supposed anywhere because it's used for internal gevent signaling.

The test here wants to test that:
- We have a pool with 1 connection
- We try to get 2 connections, which would block

To test this without relaying on gevent details I added `timeout` parameter. Without `timeout` the call blocks forever. With a timeout when the pool is deplated it will raise an exception `ConnectionPoolTimeoutError` which is not gevent specific.